physics/body_centred_body_direction_dynamic_frame_body.hpp

```diff
@@ -100,12 +113,17 @@ AcceleratedRigidMotion<InertialFrame, ThisFrame>
 BodyCentredBodyDirectionDynamicFrame<InertialFrame, ThisFrame>::
 MotionOfThisFrame(Instant const& t) const {
   DegreesOfFreedom<InertialFrame> const primary_degrees_of_freedom =
-      primary_trajectory_->EvaluateDegreesOfFreedom(t);
+      primary_trajectory_()->EvaluateDegreesOfFreedom(t);
   DegreesOfFreedom<InertialFrame> const secondary_degrees_of_freedom =
       secondary_trajectory_->EvaluateDegreesOfFreedom(t);
 
   Vector<Acceleration, InertialFrame> const primary_acceleration =
-      ephemeris_->ComputeGravitationalAccelerationOnMassiveBody(primary_, t);
+      // TODO(egg): eventually we want to add the intrinsic acceleration here.
+      primary_ == nullptr
```

Comment:
This (rather ugly) test could be avoided by having another functor `compute_gravitational_acceleration_on_primary` and setting it in the constructor.